### PR TITLE
docs($compileProvider): more information on debugInfoEnabled

### PR DESCRIPTION
**Pull request by oBusk**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Improves documentation about what `$compileProvider.debugInfoEnabled(true)` does.

**Other information**:

Based on [compile.js](https://github.com/angular/angular.js/blob/b4d1e5e492df43c49c9e4215ea75d9155fd8ba88/src/ng/compile.js#L1819...L1851) source code.

Please feel free to improve the wording or come with feedback.

Related: #16155

### CODE REVIEW

File: src/ng/compile.js [hunk 1]
```diff
@@ -1387,7 +1387,11 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
    * binding information and a reference to the current scope on to DOM elements.
    * If enabled, the compiler will add the following to DOM elements that have been bound to the scope
    * * `ng-binding` CSS class
+   * * `ng-scope` and `ng-isolated-scope` CSS classes
    * * `$binding` data property containing an array of the binding expressions
+   * * `scope()` and `isolateScope()` allowing access to scope from DOM Element
```
Comment (gkalpak) [inline src/ng/compile.js]: It doesn't add the methods. The methods are always present. It adds the `$scope` or `$isolateScope` (or `$isolateScopeNoTemplate` data properties (that can be used by the `.scope()` and `.isolateScope()` methods).

**Review by gkalpak — CHANGES_REQUESTED** [review]

**Comment by oBusk** [comment]
@gkalpak I've ammended according to your comments. Thanks for the feedback.

File: src/ng/compile.js [hunk 2]
```diff
@@ -1387,7 +1387,12 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
    * binding information and a reference to the current scope on to DOM elements.
    * If enabled, the compiler will add the following to DOM elements that have been bound to the scope
    * * `ng-binding` CSS class
+   * * `ng-scope` and `ng-isolated-scope` CSS classes
    * * `$binding` data property containing an array of the binding expressions
+   * * `$scope`, `$isolateScope` or `$isolateScopeNoTemplate` data properties which are accessable via the
```
Comment (gkalpak) [inline src/ng/compile.js]: This is too much implementation details imo. Please change to something like:

> Data properties used by the {@link angular.element#methods `scope()`/`isolateScope()` methods} to return the element's scope.

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: src/ng/compile.js (hunk 2 above)
Comment (oBusk) [inline src/ng/compile.js]: That is a very good point

**Pull request closed without merging**